app/models/manageiq/providers/openstack/inventory/collector/target_collection.rb

```diff
@@ -272,6 +274,10 @@ def infer_related_ems_refs!
       infer_related_orchestration_stacks_ems_refs_db!
       infer_related_orchestration_stacks_ems_refs_api!
     end
+    unless references(:cloud_volumes).blank?
```

Comment:
maybe we should move this up before Vms? And call ```target.manager_refs_by_association_reset``` so we do full scan of the attached Vms too?

Reply:
`target.manager_refs_by_association_reset` gets called once at the end of initialization after `parse_targets!` completes. Are you saying this would also need to be in the `parse_targets!` loop?

Reply:
Hm, lets probably keep it as you have it now. I think that VM will get all of its relations for it's own event.